pandas/core/resample.py

```diff
@@ -853,13 +853,33 @@ def var(self, ddof=1, *args, **kwargs):
 
     @Appender(GroupBy.size.__doc__)
     def size(self):
-        # It's a special case as higher level does return
-        # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
             from pandas import Series
 
-            result = Series([], index=result.index, dtype="int64")
+            if self._selected_obj.ndim == 1:
+                name = self._selected_obj.name
+            else:
+                name = None
+            result = Series([], index=result.index, dtype="int64", name=name)
```

Comment:
use self._selected_obj.__class__

Reply:
This changes does not work: it breaks unit tests where used with `DataFrame` with:

> TypeError: __init__() got an unexpected keyword argument 'name'

Reply:
Is the idea is to keep compatibility third party libraries which would ship alternative classes to `DataFrame`/`Series` (returning the same type)? This would work OK when `self._selected_obj.ndim == 1` but it does not work for other `self._selected_obj.ndim > 1`: this is because this function is expected to return an object with `ndim == 1`. I'd need a way to get the related type where `ndim==1`. Is there a blessed way to do that?

Reply:
I used `self._selected_obj.__class__ when `ndim==1`and `Series` otherwise.